ui/prometheus-plugin/src/utils/utils.ts

```diff
@@ -68,12 +68,16 @@ export const parseTemplateVariables = (text: string) => {
 export type SeriesLabels = Record<string, string>;
 
 /*
- * Formatter used for series name display in legends and tooltips
- * Regex replaces label {{ name }} with resolved label value
+ * Formatter used for series name display in legends and tooltips.
+ * Regex replaces label {{ name }} with resolved label value.
+ * If no resolved value, return empty string instead of the token inside double curly braces.
  */
 export function formatSeriesName(inputFormat: string, seriesLabels: SeriesLabels): string {
   const resolveLabelsRegex = /\{\{\s*(.+?)\s*\}\}/g;
-  return inputFormat.replace(resolveLabelsRegex, (_, g) => (seriesLabels[g] ? seriesLabels[g] : g));
+  return inputFormat.replace(resolveLabelsRegex, (_match, token) => {
+    const resolvedValue = seriesLabels[token] ?? '';
+    return resolvedValue.toString();
```

Comment:
i think you can remove `toString` here since seriesLabels[token] returns a string, right?

Reply:
I added it since I made a unit test where what was passed in wasn’t  valid but yeah I guess it wouldn’t ever actually happen, especially since this is only used in the Prometheus plugin currently